lib/kb_pickaxe/kb_pickaxeImpl.pm

```diff
@@ -243,7 +243,10 @@ sub runpickaxe
         print "generating novel compounds based on enzymatic reaction rules for $gen generations\n";
         $command .= ' -C /kb/dev_container/modules/Pickaxe/MINE-Database/minedatabase/data/EnzymaticCoreactants.tsv -r /kb/dev_container/modules/Pickaxe/MINE-Database/minedatabase/data/EnzymaticReactionRules.tsv --bnice';
 
-    } else{
+    } elsif ($params->{rule_set} eq 'retro_rules_2') {
+        print "generating novel compounds based on retro_rules_2 reaction rules for $gen generations\n";
+        system ("python3 /kb/dev_container/modules/Pickaxe/MINE-Database/minedatabase/pickaxe.py -C /kb/module/data/EnzymaticCoreactants.tsv -r /kb/module/data/retro_rules_2.tsv --bnice -q -g $gen -c /kb/module/work/tmp/inputModel.tsv -o /kb/module/work/tmp");
+    } else {
```

Comment:
As more rules sets get added, this will get more and more unwieldy. You might want to just define $coreactant_path and $rule_path in these `elsif`s and have a single system call afterward that uses those variables

Reply:
agree